app/code/Magento/Sales/Model/AdminOrder/Create.php

```diff
@@ -2094,6 +2103,47 @@ private function removeTransferredItems(): void
         }
     }
 
+    /**
+     * Restore items that were transferred from ordered items to their original sources (cart, wishlist, ...)
+     *
+     * @param string $area
+     * @param \Magento\Quote\Model\Quote\Item[]|\Magento\Wishlist\Model\Item[] $items
+     * @param \Magento\Catalog\Model\Product|null $product Product
+     * @return bool
+     */
+    private function restoreTransferredItem($area, $items, $product = null): bool
+    {
+        $transferredItems = $this->_session->getTransferredItems() ?? [];
+        if (!isset($transferredItems[$area])) {
+            return false;
+        }
+        $itemToRestoreId = null;
+        switch ($area) {
```

Comment:
Instead of a switch case, I would like to suggest using a composite approach like below:

https://github.com/magento-commerce/magento2ce/blob/b98deb3fea25cb823c25b005d3ea22ab18c45176/app/code/Magento/Ui/etc/di.xml#L405

Reply:
I'm getting a 404 on your link.

Reply:
The above link was from the private repo, you can use the below one for reference:

https://github.com/magento/magento2/blob/b98deb3fea25cb823c25b005d3ea22ab18c45176/app/code/Magento/Ui/etc/di.xml#L405